README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->
[![Travis-CI Build Status](https://travis-ci.org/xmarquez/QuickUDS.svg?branch=master)](https://travis-ci.org/xmarquez/QuickUDS)

QuickUDS
========

This `R` package contains convenience functions to extend the [Unified Democracy Scores of Pemstein, Meserve, and Melton](http://www.unified-democracy-scores.org/). These are basically wrappers for a couple of functions in the package [mirt](https://cran.r-project.org/web/packages/mirt/), which contains powerful facilities for factor analysis, and which you will probably want to use directly to do anything beyond replicating and extending the UD scores with the data provided by this package.

This package also includes an "[extended UDS](reference/extended_uds.html)" dataset with latent democracy scores going back to the 19th century (and before, in some cases) and a replication dataset with 74 different democracy measurements from 30 different projects to measure democracy. Both datasets are fully documented. The extended scores (and their generation) are described in more detail in Marquez, Xavier, "A Quick Method for Extending the Unified Democracy Scores" (March 23, 2016). Available at SSRN: <http://ssrn.com/abstract=2753830>. A more informal description of the extended UD scores is also found in [this blogpost](http://abandonedfootnotes.blogspot.co.nz/2016/03/artisanal-democracy-data-quick-and-easy.html).

Installation
------------

You can install QuickUDS from github with:

``` r
# install.packages("devtools")
devtools::install_github("xmarquez/QuickUDS/")
```

The main datasets in the package are `extended_uds` and `democracy`. Access their documentation in `R` by typing `?extended_uds` or `?democracy`.

Documentation for all functions and datasets in this package is always available [at this website](reference/index.html). Click the [reference](reference/index.html) link for more information.

For non-R users, there are csv versions of all datasets available [here](https://github.com/xmarquez/QuickUDS/tree/master/csvs-and-pdfs).

Usage
-----

See the [package vignettes](articles/index.html) for detailed examples of how to generate the Extended Unified Democracy Scores dataset or similar measures. The vignette [Generating the Extended Unified Democracy Scores](articles/Generating_the_extended_UDS.html) also plots the calculated scores for every country.

Citation
--------

If you find this package useful in your work, please cite:

Pemstein, Daniel, Stephen Meserve, and James Melton. 2010. Democratic Compromise: A Latent Variable Analysis of Ten Measures of Regime Type. Political Analysis 18 (4): 426-449.

As well as:

Marquez, Xavier, "A Quick Method for Extending the Unified Democracy Scores" (March 23, 2016). Available at SSRN: <http://ssrn.com/abstract=2753830>


NEWS.md
# QuickUDS 0.2.3

* Incorporated latest version of Boix, Miller, and Rosato's (2012) dataset (v. 3.0) as well as a new Support Vector Machine index by Grundler and Krieger in the construction of the extended UDS.

* Revised documentation and vignettes.

* Added the `polity_ccode` code to the datasets again.

# QuickUDS 0.2.2

* Added a `NEWS.md` file to track changes to the package.

* Package now uses the latest version of Freedom House data (to 2017, report year 2018) and of the WGI's voice and acocountability index (to 2016).





R/extended_uds_documentation.R
#' Extended UDS
#'
#' A dataset extending the Unified Democracy Scores of Pemstein, Meserve, and
#' Melton (2010) to the 19th century (and sometimes before), updating it with
#' 2013-2016 data, and calculating scores for countries not in the official UD
#' release. Cite both Pemstein, Meserve, and Melton 2010 as well as Marquez
#' 2016.
#'
#' @section Overview:
#'
#'   This dataset contains an extension of the Unified Democracy Scores of
#'   Pemstein, Meserve, and Melton (2010). PMM 2010 use a latent variable
#'   approach to combine diverse measurements of democracy for a broad panel of
#'   countries from 1946 to 2012; this dataset extends the scores by using more
#'   democracy measures with broader temporal and spatial coverage, generating
#'   democracy scores for 26,258 country-years (248 distinct countries and
#'   territories), including 18,345 country-years for independent, sovereign
#'   countries from 1816 to 2016.
#'
#'   The measurement of democracy is complicated and controversial. The extended
#'   UDS makes few judgments about what measures of democracy should be used to
#'   generate latent democracy scores, as long as they have been used in
#'   scholarly work. It uses dichotomous, trichotomous, ordinal, and continuous
#'   indices; indices that focus primarily on the "competition" and indices that
#'   focus on the "participation" dimension of democracy; "thick" indices that
#'   attempt to measure a wide variety of characteristics plausibly attributed
#'   to democracy; and "minimalist" indexes that restrict themselves to the bare
#'   minimum of competition. Nevertheless, most of these indexes agree that
#'   democracy has something to do with competition and participation, even if
#'   they weight these dimensions somewhat differently, and even if they include
#'   other things, such as civil rights.
#'
#'   The latent variable score is calculated using the following variables in
#'   the [democracy] dataset: "pmm_arat", "blm", "bmr_democracy",
#'   "bnr_extended", "pmm_bollen", "doorenspleet",
#'   "wgi_democracy","fh_total_reversed", "fh_electoral",
#'   "gwf_democracy_extended", "pmm_hadenius", "kailitz_tri", "lexical_index",
#'   "mainwaring", "magaloni_democracy_extended", "pmm_munck", "pacl", "PEPS1v",
#'   "pitf", "polity2", "reign_democracy", "polyarchy_original_contestation",
#'   "prc", "svolik_democracy", "ulfelder_democracy_extended",
#'   "utip_dichotomous_strict", "v2x_polyarchy", "vanhanen_democratization",
#'   "wth_democ1". Please refer to the documentation of [democracy] for more
#'   details on these measures of democracy, and to Marquez 2016 for more
#'   details on the construction of the extended latent variable model used to
#'   generate the scores.
#'
#' @template standard-variables
#'
#' @template latent-variables
#'
#' @references
#'
#' Marquez, Xavier. 2016. A Quick Method for Extending the Unified Democracy
#' Scores (March 23, 2016). Available at SSRN:
#' \url{http://ssrn.com/abstract=2753830}
#'
#' Pemstein, Daniel, Stephen Meserve, and James Melton. 2010. Democratic
#' Compromise: A Latent Variable Analysis of Ten Measures of Regime Type.
#' Political Analysis 18 (4): 426-449. DOI:10.1093/pan/mpq020
#'
#' @family democracy
#' @author Xavier Marquez, Political Science and International Relations
#'   Programme, Victoria University of Wellington, marquez.x@gmail.com
"extended_uds"

